Shop_Unity_base/Assets/Scripts/Shop/View/ShopView.cs

```diff
@@ -28,92 +25,103 @@ public class ShopView : MonoBehaviour
+            ShopModel.OnBuyEvent += OnShopUpdated;
+            ShopModel.OnSellEvent += OnShopUpdated;
+
+            PopulateItemIconView();
             InitializeButtons();
         }
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  RepopulateItems()
-        //------------------------------------------------------------------------------------------------------------------------        
-        //clears the icon gridview and repopulates it with new icons (updates the visible icons)
-        private void RepopulateItemIconView() {
+        private void OnShopUpdated(Item item, Customer customer)
+        {
+            RepopulateItemIconView();
+        }
+
+        private void OnDestroy()
+        {
+            ShopModel.OnBuyEvent -= OnShopUpdated;
+            ShopModel.OnSellEvent -= OnShopUpdated;
+        }
+
+        // Clears the icon gridview and repopulates it with new icons (updates the visible icons)
+        private void RepopulateItemIconView()
+        {
             ClearIconView();
             PopulateItemIconView();
         }
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  PopulateItems()
-        //------------------------------------------------------------------------------------------------------------------------        
-        //adds one icon for each item in the shop
-        private void PopulateItemIconView() {
-            foreach (Item item in shopModel.GetItems()) {
+        // Adds one icon for each item in the shop
+        private void PopulateItemIconView()
+        {
+            foreach (Item item in shopModel.GetItems())
                 AddItemToView(item);
-            }
         }
 
-        //------------------------------------------------------------------------------------------------------------------------
-        //                                                  ClearIconView()
-        //------------------------------------------------------------------------------------------------------------------------        
-        //remove all existing icons in the gridview
-        private void ClearIconView() {
+        // Remove all existing icons in the gridview
+        private void ClearIconView()
+        {
             Transform[] allIcons = itemLayoutGroup.transform.GetComponentsInChildren<Transform>();
-            foreach (Transform child in allIcons) {
-                if (child != itemLayoutGroup.transform) {
+            foreach (Transform child in allIcons)
+            {
+                if (child != itemLayoutGroup.transform)
                     Destroy(child.gameObject);
-                }
-             }
+            }
```

Comment:
At this point, if you're removing braces for 'if' statements, why not remove them for single-line 'for' statements too?

Reply:
I do that as well, but only for an actual single line `for` statement.
This is bad for me: 
```cs
foreach (Transform child in allIcons)
    if (child != itemLayoutGroup.transform)
        Destroy(child.gameObject);
```
And you could keep nesting them, makes it harder to read.
If the body of an `if` or `for` is longer than 1 line, I prefer this:
```cs
foreach (Transform child in allIcons) 
{
    if (child != itemLayoutGroup.transform)
        Destroy(child.gameObject);
}
```